lib/CPython/Simple.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module CPython.Simple where
+
+import CPython.Simple.Instances
+
+import Control.Exception (catch, SomeException, throwIO)
+import Data.Semigroup ((<>))
+import Data.Text (Text)
+import qualified Data.Text as T
+import System.IO (hPutStrLn, stderr)
+
+import qualified CPython as Py
+import qualified CPython.Protocols.Object as Py
+import qualified CPython.Types as Py
+import qualified CPython.Types.Module as Py
+import qualified CPython.Types.Tuple as Py (toTuple)
+import qualified CPython.Types.Dictionary as PyDict
+
+data Arg = forall a. ToPy a => Arg a
+
+instance ToPy Arg where
+  toPy (Arg a) = toPy a
+
+arg :: ToPy a => a -> Arg
+arg = Arg
+
+initialize :: IO ()
+initialize = Py.initialize `catch`
+  \(_ :: SomeException) -> error $ "Couldn't initialize.\nMaybe <Python.h> is missing?"
+
+importModule :: Text -> IO Py.Module
+importModule module_ = Py.importModule module_ `catch`
+  \(e :: SomeException) -> do
+    hPutStrLn stderr $
+      "Error: Couldn't import Python module named `" <> T.unpack module_ <> "`\n" <>
+      "Maybe it's a typo, or `" <> T.unpack module_ <> "` isn't installed?\n" <>
+      "Check with `python -m " <> T.unpack module_ <> "`"
+    throwIO e
+
+call
+  :: FromPy a
+  => Text -- ^ module name
+  -> Text -- ^ function name
+  -> [Arg] -- ^ args
+  -> [(Text, Arg)] -- ^ kwargs
+  -> IO a
+call moduleName func args kwargs = (do
+  module_ <- importModule moduleName
+  pyFunc <- Py.getAttribute module_ =<< Py.toUnicode func
+  pyArgs <- mapM toPy args
+  pyArgsTuple <- Py.toTuple pyArgs
+  pyKwargs <- toPyKwargs kwargs
+  result <- Py.call pyFunc pyArgsTuple pyKwargs
+  fromPy result) `catch`
+    \(e :: SomeException) -> do
+      hPutStrLn stderr $
```

Comment:
generally speaking I do not like the idea of putting error messages on stderr from a library